pages/api/user/[userId]/index.js

```diff
@@ -0,0 +1,43 @@
+import dbConnect from "@/db/dbConnect";
+
+import { getToken } from "next-auth/jwt";
+import User from "@/db/models/User";
+
+export default async function handler(request, response) {
+  const { userId } = request.query;
+  const token = await getToken({
+    req: request,
+    secret: process.env.NEXTAUTH_SECRET,
+  });
+  if (!token) {
+    return response.status(401).json({ error: "Not authenticated" });
+  }
+
+  try {
+    await dbConnect();
+    if (request.method === "PUT") {
+      const userToToggleDarkMode = await User.findById(userId);
+      userToToggleDarkMode.isDarkMode = request.body.isDarkMode;
+
+      const updatedUser = await User.findByIdAndUpdate(
+        userId,
+        userToToggleDarkMode,
+        {
+          new: true,
+          runValidators: true,
+        }
+      );
```

Comment:
you don't need the entire user object in MongoDB to change one value. Instead, you can just pass the elements that should change, everything else is kept as is:
```js
await User.findByIdAndUpdate(userId, { isDarkMode: request.body.isDarkMode })
```